revscoring/features/revision.py

```diff
@@ -0,0 +1,235 @@
+
+def process_uppercase_chars(revision_text):
+    return sum(c.lower() != c for c in revision_text)
+
+uppercase_chars = Feature("revision.uppercase_chars",
+                          process_uppercase_chars,
+                          returns=int, depends_on=[revision.text])
+
+proportion_of_uppercase_chars = uppercase_chars / modifiers.max(chars, 1)
+
+################################## Words #######################################
+
+def process_words(revision_words):
+    return len(revision_words)
+
+words = Feature("revision.words", process_words,
+                returns=int, depends_on=[revision.words])
+
+def process_badwords(is_badword, revision_words):
+    return sum(is_badword(word) for word in revision_words)
+
+badwords = Feature("revision.badwords", process_badwords,
+                   returns=int,
+                   depends_on=[is_badword, revision.words])
+
+proportion_of_badwords = badwords / modifiers.max(words, 1)
+
+def process_misspellings(is_misspelled, revision_words):
+    return sum(is_misspelled(word) for word in revision_words)
+
+misspellings = Feature("revision.misspellings", process_misspellings,
+                       returns=int,
+                       depends_on=[is_misspelled, revision.words])
+
+proportion_of_misspellings = badwords / modifiers.max(words, 1)
+
+
+################################ Parse tree ####################################
+
+def process_level_1_headings(headings):
+    return sum(h.level==1 for h in headings)
+
+level_1_headings = \
+        Feature("revision.level_1_headings", process_level_1_headings,
+                returns=int, depends_on=[revision.headings])
+                           
+def process_level_2_headings(headings):
+   return sum(h.level==2 for h in headings)
+
+level_2_headings = \
+        Feature("revision.level_2_headings", process_level_2_headings,
+                returns=int, depends_on=[revision.headings])
+
+def process_level_3_headings(headings):
+    return sum(h.level==3 for h in headings)
+
+level_3_headings = \
+        Feature("revision.level_3_headings", process_level_3_headings,
+                returns=int, depends_on=[revision.headings])
+
+def process_level_4_headings(headings):
+    return sum(h.level==4 for h in headings)
+
+level_4_headings = \
+        Feature("revision.level_4_headings", process_level_4_headings,
+                returns=int, depends_on=[revision.headings])
+
+def process_level_5_headings(headings):
+    return sum(h.level==5 for h in headings)
+
+level_5_headings = \
+        Feature("revision.level_5_headings", process_level_5_headings,
+                returns=int, depends_on=[revision.headings])
+
+def process_level_6_headings(headings):
+    return sum(h.level==5 for h in headings)
+
+level_6_headings = \
+        Feature("revision.level_6_headings", process_level_6_headings,
+                returns=int, depends_on=[revision.headings])
+
+def process_infonoise(is_stopword, stem_word, content_words):
```

Comment:
Why are the non-stopwords called noise? Isn't it the opposite (the stopwords are the noise)?